src/components/memorygame/MemoryGameFlippableCard/index.jsx

```diff
@@ -0,0 +1,68 @@
+import React from "react";
+import ReactCardFlip from "react-card-flip";
+import { Card } from "antd";
+import questionMark from "../../../images/memoryGameFrontImage.png";
+import "./index.css";
+
+// for cardStates example check src/components/memorygame/MemoryGameBoard/index.jsx
+// FlippableCard is being used in a map function. So targetState refers to each iteration of map
+// Mapped array is cardStates.
+// selectedCards is an array of clicked card states(objects).
+const FlippableCard = ({
+  targetState,
+  cardStates,
+  setCardStates,
+  selectedCards,
+  setSelectedCards,
+}) => {
+  // handle click flips card
+  // if the card is not in selectedCards, adds it to selectedCards
+  const handleClick = (clickedCardState) => {
+    const copyOfCardStates = [...cardStates];
+    // current index is needed to keep the order of cards displayed
+    // because goal is to make user remember the spots of previous cards
+    const clickedCardIndex = cardStates.findIndex(
+      (state) => state.cardKey === clickedCardState.cardKey
+    );
+    const targetCardState = copyOfCardStates[clickedCardIndex];
+    targetCardState.isFlipped = true;
+    setCardStates(() => [...copyOfCardStates]);
+    // prevents user to use same card multiple times
```

Comment:
does your game handle unflipping if a card is selected?

i pick card 0

then i click card 0 again (its the first selected card)

does it unflip?

you don't have to do this, i'm just asking, but then you also probably want flipped cards to have a not-allowed over it if you didnt already

Reply:
I thought it is for keeping correctly matching cards, if it is true match user can't unflip.

Reply:
Unflipping is done automatically when user makes a wrong match. So if you select same card twice it doesnt unflip. I added conditional class selection to implement not allowed cursor. PTAL